src/daemon/daemon.cpp

```diff
@@ -1241,9 +1241,40 @@ try // clang-format on
         }
 
         auto& vm = it->second;
+        auto& vm_specs = vm_instance_specs[name];
 
         if (vm->current_state() == mp::VirtualMachine::State::running)
         {
+            // If the given path is relative, we need to resolve the fill
+            // absolute path. For this, we execute the command 'pwd'.
+            if (QDir(QString::fromStdString(target_path)).isRelative())
+            {
+                mp::SSHSession session{vm->ssh_hostname(), vm->ssh_port(), vm_specs.ssh_username, *config->ssh_key_provider};
+
+                std::string home;
+
+                auto proc = session.exec("pwd");
+                if (proc.exit_code() != 0)
+                {
+                    auto error_msg = proc.read_std_error();
+                    mpl::log(
+                        mpl::Level::warning, category,
+                        fmt::format("failed to run 'pwd', error message: '{}'", mp::utils::trim_end(error_msg)));
+                    home = std::string{};
+                }
+
+                home = proc.read_std_output();
+                mp::utils::trim_end(home);
+
+                target_path = mp::utils::normalize_absolute_path(home + '/' + target_path);
+                mpl::log(mpl::Level::info, category, fmt::format("home directory: {}, target_path: {}", home, target_path));
+            }
+            else
+            {
+                target_path = mp::utils::normalize_absolute_path(target_path);
+                mpl::log(mpl::Level::info, category, fmt::format("target_path: {}", target_path));
+            }
+
```

Comment:
Since this code is repeated below, could it be extracted into a helper function, to avoid duplication? Something returning the resolved and normalized `target_path` from the given the VM name and `target_path` input.

Reply:
Nice, I'll do it.